### DISCUSSION TITLE
Evaluate tool: can't see Client Roles on role select field

### DISCUSSION THREAD

**Discussion by rcapp** [keycloak/keycloak#16869] [Archive]
Hello, everyone! 

I'm trying to use the Evaluate tool to check if everything is fine with my user permissions. However, I'm not able to select the roles I have created for my client. The only options shown are `default-roles` and `uma_authorization`, plus Realm Roles.  

What am I missing here? 

Also, in the documentation section there isn't much information about how the tool work and how the evaluation really find out if the your permit or deny. Are the scopes we select used to ask "evaluate user *supposing* they have this scope" or "evaluate *if* user are in these scopes *and* can access resources related to them".

At last: the documentation screenshot shows `Role` as not mandatory field,  but the actual application marks it as mandatory. Which one is the correct?

Thank you in advance!

**Comment by dantemessy** [comment]
Any updates?

**Comment by monzonj** [comment]
I'm also having this issue. Only realm roles are selectable.
this used to work and has changed recently.
